scripts/dumpapp

```diff
@@ -53,19 +40,26 @@ def main():
   except KeyboardInterrupt:
     sys.exit(1)
 
-def read_frames(conn):
+def read_frames(sock):
   while True:
     # All frames have a single character code followed by a big-endian int
-    code = read_input(conn, 1, 'code')
-    n = struct.unpack('!L', read_input(conn, 4, 'int4'))[0]
+    code = read_input(sock, 1, 'code')
+    n = struct.unpack('!L', read_input(sock, 4, 'int4'))[0]
 
     if code == b'1':
       if n > 0:
-        sys.stdout.buffer.write(read_input(conn, n, 'stdout blob'))
+        sys.stdout.buffer.write(read_input(sock, n, 'stdout blob'))
     elif code == b'2':
       if n > 0:
-        sys.stderr.buffer.write(read_input(conn, n, 'stderr blob'))
+        sys.stderr.buffer.write(read_input(sock, n, 'stderr blob'))
         sys.stderr.buffer.flush()
+    elif code == b'_':
```

Comment:
We can't directly "connect" stdin from the terminal to the app because it means we will perform a read that wasn't requested.  This manifests itself in strange ways for non-trivial scripts such as:

```
$ echo -en "123" | (head -c 1; tail -c 1); echo
13
```

Both the head and tail invocation are given the same fd (/dev/stdin is fixed at this point) and both open the fd, read what they are told to read, then exit.  In this way, I can instruct head to read just 1 byte, leaving the remaining two on the stream for tail to skip one and print the last.  If head was coded as you suggest dumpapp to be coded, then the fd will be exhausted immediately and forwarded to head.  Tail would then print nothing.

This framing protocol design allows us instead to "connect" the fds perfectly, transferring each individual read call from the dumper plugin to a read call on the dumpapp script's stdin as it would work in a pipe situation normally.